### PR TITLE
STY: PEP8 format and test_format.py

### PR DESCRIPTION
**Pull request by ElDeveloper**

### CODE REVIEW

**Comment by emperor-helper** [comment]
The test build for this pull request can be found here: http://emperor.colorado.edu/pull_381/make_emperor/

File: emperor/format.py [hunk 1]
```diff
@@ -1,76 +1,85 @@
...
-
-    Formats the output of qiime.parse.parse_coords_file into javascript variable
-    declarations.
+    Parameters
+    ----------
+    header : list of str
+        sample names for the pcoa file 1-D array
+    coords : array-like
+        coordinates of the PCoA file, 2-D array
+    eigvals : array-like
```
Comment (mortonjt) [inline emperor/format.py]: array_like

File: emperor/format.py [hunk 2]
```diff
@@ -1,76 +1,85 @@
...
+    string
+        javascript representation of the PCoA data inputed, contains a list of
+        spheres, list of ellipses (if coords_low and coords_high are present)
+        and several setup variables.
+
+    Formats the output of qiime_backports.parse.parse_coords_file into
+    javascript variable declarations.
     """
-    js_pcoa_string = ''
+    eigvals *= 1
```
Comment (mortonjt) [inline emperor/format.py]: eigvals not being used?

File: emperor/format.py [hunk 3]
```diff
@@ -298,114 +359,140 @@ def format_vectors_to_js(mapping_file_data, mapping_file_headers, coords_data,
...
+        whether the plot has ellipses or not
+    has_vectors : bool, optional
+        whether the plot has vectors or not
+    has_edges : bool, optional
+        whether the plot has edges between samples (comparison plot)
+
+    Returns
+    -------
+    str
+        Fromatted footer string
```
Comment (mortonjt) [inline emperor/format.py]: typo! should be "formatted"

File: emperor/format.py [hunk 4]
```diff
@@ -151,32 +173,41 @@ def format_pcoa_to_js(header, coords, eigvals, pct_var, custom_axes=[],
...
+    mapping_file_data : list of str
+        contents of the mapping file
+    mapping_file_headers : list of list of str
+        headers of the mapping file
+    columns : list of str
+        valid columns to use, usually a subset of mapping_file_headers
+
+    Returns
+    -------
+    string
```
Comment (mortonjt) [inline emperor/format.py]: returns `string` or `str`

The other return statements are returning `str`

File: emperor/format.py [hunk 5]
```diff
@@ -298,114 +359,140 @@ def format_vectors_to_js(mapping_file_data, mapping_file_headers, coords_data,
...
-    AssertionError if the coords_data and coords_headers don't have the same
-    length.
-    AssertionError if the number of clones doesn't concord with the samples
-    being presented.
-
+    Parameters
+    ----------
+    coords_data : array-like
+        numpy array with the replicated coordinates
+    cooreds_headers : list of str
```
Comment (mortonjt) [inline emperor/format.py]: needs to be `corrds_headers`

**Comment by ElDeveloper** [comment]
Ok, this should be ready to be reviewed/merged, I've taken a pass at a few minor missing things. Please review carefully :+1:!

File: emperor/format.py [hunk 6]
```diff
@@ -80,69 +94,82 @@ def format_pcoa_to_js(header, coords, eigvals, pct_var, custom_axes=[],
...
+                         all_coords))
 
     # write the values for all the ellipses
-    js_pcoa_string += '\nvar g_ellipsesDimensions = new Array();\n'
+    pcoa_str.append('\nvar g_ellipsesDimensions = new Array();\n')
     if coords_low is not None and coords_high is not None:
         for s_header, s_coord, s_low, s_high in zip(header, coords, coords_low,
-            coords_high):
+                                                    coords_high):
             delta = abs(s_high-s_low)
```
Comment (antgonza) [inline emperor/format.py]: Spaces between -

File: emperor/format.py [hunk 7]
```diff
@@ -80,69 +94,82 @@ def format_pcoa_to_js(header, coords, eigvals, pct_var, custom_axes=[],
...
-    js_pcoa_string += 'var g_yAxisLength = %f;\n' % (abs(max_y)+abs(min_y))
-    js_pcoa_string += 'var g_zAxisLength = %f;\n' % (abs(max_z)+abs(min_z))
-    js_pcoa_string += 'var g_xMaximumValue = %f;\n' % (max_x)
-    js_pcoa_string += 'var g_yMaximumValue = %f;\n' % (max_y)
-    js_pcoa_string += 'var g_zMaximumValue = %f;\n' % (max_z)
-    js_pcoa_string += 'var g_xMinimumValue = %f;\n' % (min_x)
-    js_pcoa_string += 'var g_yMinimumValue = %f;\n' % (min_y)
-    js_pcoa_string += 'var g_zMinimumValue = %f;\n' % (min_z)
-    js_pcoa_string += 'var g_maximum = %f;\n' % maximum
+            pcnts = ["'P%d': %f" % (i+1, s_coord[i]) for i in
```
Comment (antgonza) [inline emperor/format.py]: spaces between +

File: emperor/format.py [hunk 8]
```diff
@@ -80,69 +94,82 @@ def format_pcoa_to_js(header, coords, eigvals, pct_var, custom_axes=[],
...
+    # create three vars, pc1, pc2 and pc3 if no custom_axes are passed, then
+    # use the values of the percent explained by the PCoA; if custom_axes are
+    # passed use as many as you can (since customs axes can be either [0, 1, 2,
+    # 3])
     for i in range(0, 3):
         try:
-            js_pcoa_string += 'var g_pc%dLabel = \"%s\";\n' % (i+1,
-                custom_axes[i])
-            offset+=1 # offset will help us retrieve the correct pcoalabels val
+            pcoa_str.append('var g_pc%dLabel = \"%s\";\n' % (i+1,
```
Comment (antgonza) [inline emperor/format.py]: spaces between +

File: emperor/format.py [hunk 9]
```diff
@@ -80,69 +94,82 @@ def format_pcoa_to_js(header, coords, eigvals, pct_var, custom_axes=[],
...
+                                                             custom_axes[i]))
+            # offset will help us retrieve the correct pcoalabels val
+            offset += 1
         except:
             # if there are custom axes then subtract the number of custom axes
-            js_pcoa_string += 'var g_pc%dLabel = \"PC%d (%.2f %%)\";\n' %\
-                (i+1, i+1-offset, pcoalabels[i-offset])
-    js_pcoa_string += 'var g_number_of_custom_axes = %d;\n' % offset
+            pcoa_str.append('var g_pc%dLabel = \"PC%d (%.2f %%)\";\n' %
+                            (i+1, i+1-offset, pcoalabels[i-offset]))
```
Comment (antgonza) [inline emperor/format.py]: spaces between + and -

File: tests/all_tests.py [hunk 10]
```diff
@@ -209,7 +213,7 @@ def main():
     # In case there were no failures of any type, exit with a return code of 0
     return_code = 1
     if (len(bad_tests) == 0 and len(missing_application_tests) == 0 and
-        script_usage_failures == 0 and javascript_tests_passed):
+       script_usage_failures == 0 and javascript_tests_passed):
```
Comment (antgonza) [inline tests/all_tests.py]: The space removed was actually needed.

File: emperor/core.py [hunk 11]
```diff
@@ -14,26 +14,25 @@
...
 
 from emperor.format import (format_mapping_file_to_js, format_pcoa_to_js,
                             format_taxa_to_js, format_vectors_to_js,
                             format_comparison_bars_to_js,
-                            EMPEROR_HEADER_HTML_STRING,
                             format_emperor_html_footer_string)
+from emperor._format_strings import EMPEROR_HEADER_HTML_STRING
 
 # we are going to use this remote location to load external resources
 RESOURCES_URL = 'http://emperor.colorado.edu/master/make_emperor/emperor_outpu\
```
Comment (antgonza) [inline emperor/core.py]: added https://github.com/biocore/emperor/issues/382

**Comment by ElDeveloper** [comment]
Ok, the build is passing, 100% PEP8 compliant!  :+1:

**Comment by ElDeveloper** [comment]
Thanks @antgonza

**Comment by ElDeveloper** [comment]
Thanks!

**Pull request merged**